labman/gui/handlers/process_handlers/library_prep_16s_process.py

```diff
@@ -53,7 +53,13 @@ def get(self):
         tools_tm300_8 = Equipment.list_equipment(
             'tm 300 8 channel pipette head')
         tools_tm50_8 = Equipment.list_equipment('tm 50 8 channel pipette head')
-        primer_plates = Plate.list_plates(['primer'])
+
```

Comment:
I think you can use plate_types in list_plates?

Reply:
Unfortunately doesn't work. That tests against the descriptipn field in the composition table (or composition types? Can't remember off hand), and the entries in there do not provide sufficient granularity to determine whether shotgun or amplicon

Reply:
Then I'm wondering if perhaps worth fixing there vs. looping over all the plates?

Reply:
(github didn't reply to the right place via email)

That's i suspect a pretty reasonable sized data model change. This doesn't
loop over all plates either, just the primer plates

Reply:
The table type restriction ("primer" in this case) filters on this [column](https://github.com/jdereus/labman/blob/master/labman/db/support_files/db_patch.sql#L5). The distinction, as best I can tell, for amplicon or shotgun is based off this [column](https://github.com/jdereus/labman/blob/master/labman/db/support_files/db_patch.sql#L47) in a separate table `primer_set`. The `primer_set` table is not part of the constructed SQL query used by `list_plates`. `list_plates` is already a complex function. It may be possible to increase its complexity. However I'm not familiar enough with the broader data model to know whether this is an edge case in the data model (i.e., this should be in `composition_types`), because it seems weird that this information is "difficult" to get too. 

As best I can tell, this filtering we're doing here is only done in these locations. We can create a separate issue for decomposing where and how this filtering is being performed, but I would be hesitant to include it in the full launch milestone.

Reply:
After checking the code and testing, it looks like plate_types has to be an _array_ and one of [these values](https://github.com/jdereus/labman/blob/master/labman/db/composition.py#L43-L53); something like this:`list_plates(plate_types=['16S library prep'])`

Reply:
That is the wrong information and does not solve this problem. 

```python
In [6]: Plate.list_plates(["16S library prep"])
Out[6]:
[{'plate_id': 23, 'external_id': 'Test 16S plate 1'},
 {'plate_id': 29, 'external_id': 'Test 16S plate 2'},
 {'plate_id': 32, 'external_id': 'Test 16S plate 3'},
 {'plate_id': 35, 'external_id': 'Test 16S plate 4'}]
```

Reply:
Got it now, you are searching for all the primer plates that are Shotgun or Amplicon, thanks for explanation. Would it be more convenient to add that option to the list_plates method? So you can call like:
```
Plate.list_plates() # returns everything
Plate.list_plates(plate_types=['primer']) # returns all primers
Plate.list_plates(plate_types=['primer'], target='Shotgun') # returns Shotgun primers?
```
This should also avoid the duplication of the loops ...